components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -83,8 +84,13 @@ private void validate(String username, String userStoreDomain, String tenantDoma
     private List<String> getSessionsOfUser(String username, String userStoreDomain, String tenantDomain) throws
             UserSessionException {
 
-        String userId = UserSessionStore.getInstance().getUserId(username, getTenantId(tenantDomain),
-                userStoreDomain);
+        String userId;
+        if (StringUtils.isEmpty(userStoreDomain) || FrameworkConstants.FEDERATED_IDP_NAME.equals(userStoreDomain)){
+            userId = UserSessionStore.getInstance().getUserId(username, getTenantId(tenantDomain),
+                    userStoreDomain);
```

Comment:
Looking at the triggering point of this call [1], it doesn't look like a user store domain can be "FEDERATED".
Did we verify this
[1] https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/UserSessionTerminationListener.java#L134

Reply:
Basically this method would never be invoked for a federated it seems, unless we provide an API and consume the OSGI service for a federated user case.
@pamodaaw , did we do such ?

Reply:
We don't have such API for federated users which requires the OSGi service. When storing the user details in the IDN_AUTH_USER table, if the userstore domain is invalid, we assume that the user is a federated user and in the entry, the user store domain is marked as FEDERATED.